Sources/SwiftDocC/Model/Rendering/RenderContentCompiler.swift

```diff
@@ -207,6 +205,44 @@ struct RenderContentCompiler: MarkupVisitor {
         return [RenderInlineContent.strikethrough(inlineContent: strikethrough.children.reduce(into: [], { result, child in result.append(contentsOf: visit(child))}) as! [RenderInlineContent])]
     }
 
+    mutating func visitBlockDirective(_ blockDirective: BlockDirective) -> [RenderContent] {
+        switch blockDirective.name {
+        case Snippet.directiveName:
+            guard let snippetURL = blockDirective.arguments()[Snippet.Semantics.Path.argumentName],
+                  let snippetReference = resolveSymbolReference(destination: snippetURL.value),
+                  let snippetEntity = try? context.entity(with: snippetReference),
+                  let snippetSymbol = snippetEntity.symbol,
+                  let snippetMixin = snippetSymbol.mixins[SymbolGraph.Symbol.Snippet.mixinKey] as? SymbolGraph.Symbol.Snippet else {
+                return []
+            }
+
+            let docCommentContent = snippetEntity.markup.children.flatMap { self.visit($0) }
+
+            let codeContent = snippetMixin.chunks.flatMap { chunk -> [RenderContent] in
+                guard !chunk.code.isEmpty else {
+                    return []
+                }
+
+                var elements = [RenderContent]()
+
+                if let chunkName = chunk.name {
+                    elements.append(RenderBlockContent.paragraph(inlineContent: [
```

Comment:
This code path isn't taken in the tests. I think that it'd be good to add a name to the snippet in the test data to be able to assert that that content is processed as expected.

Reply:
Sure thing.

Reply:
As an aside, chunk names aren't yet authorable but I can add some more test cases once that's implemented on the SwiftPM side if that's all right?

Reply:
Yes. That sounds good to me.